docs/events/glides/com.mbta.ctd.glides.vehicle_trip_assignments.v1.mdx

```diff
@@ -0,0 +1,184 @@
+
+If Glides expects a train to turn around at a station in the middle of the line at a station where trains must loop to a different platform (Government Center Eastbound to Westbound, North Station, Kenmore, Park Street), when the train reaches the last stop, Glides will leave the train assigned to the previous trip. Glides will not unassign it from the previous trip and assign the train to the next trip until the train has actually turned around onto track that goes the other direction. This is so that if a train is unexpectedly extended (e.g. continuing from Government Center until North Station), the train won't be unassigned from a trip until it _actually_ finishes the trip.
+
+During layovers, Glides will assign a vehicle to the upcoming trip as long as Glides thinks that the vehicle will do that trip, the trip is leaving soon, and the train is at the station waiting to leave or traveling towards the station in the same direction as the upcoming trip. This is so that it's easier to provide riders with good information while a trip is waiting to start, and so that Glides can more reliably assign vehicles to trips in some situations. For example, consider the case of a westbound trip that's supposed to start at Government Center, but the train is starting at North Station because the previous trip was extended. When the train is heading back westbound, even before it reaches Goverment Center, we want the vehicle to be assigned to the westbound trip. And when the train is passing Government Center Westbound, it would be hard to tell that the train is actually in the middle of the trip and not waiting to start its trip. The most reliable way to make sure that the train is assigned to the trip at all times when it's doing it's trip is to assign it to the trip even before we think the trip has started, while it's laying over or on its way to the starting station.
+
+### Other
+
+Glides may unassign or reassign vehicles to trips at other times, too. For example, if a train unexpectedly takes a different branch, or if a train turns around in an unexpected place, or an inspector enters new information into Glides. This section is not a comprehensive spec of how Glides decides which trips to assign vehicles to.
+
+Glides may not be able to make an assignment for every train. If it doesn't recognize the trip a train is doing, it will leave the train unassigned to any trip. It will not create a new added trip just to have a trip ID to reference. Trains will only be assigned to added trips if an official has written an added trip into the trainsheet in Glides.
+
+Glides may publish different assignments in the Glides app and in this event feed, if it decides the needs of operations staff (who use the Glides app) and riders (who use data powered by this feed) are different. We expect to follow slightly different rules around layovers for in-app trip assignments.
```

Comment:
**question:** maybe out of scope, but do you think that there are any things we can say about the relationship between what shows up in the feed and what shows up in the app? I agree that in general there's no guarantee that they be 100% the same, but do you think we could say (for instance) that _if_ a train is assigned to a trip in the Glides UI, _then_ it will be assigned to that trip in the feed, but the converse isn't necessarily true? Or do you think that even that is too strong of an assertion to make?

Reply:
The two sources might assign a vehicle to two different trips, not just one has an assignment and the other doesn't, so that specific assertion wouldn't work.
We're only planning to do it at layovers, so could maybe the spec could say that?

But the whole point of this sentence is that it's _not_ making any specific assertions. They should both be correct from their own perspective, and they'll be mostly the same, so it's not like a specific assertion here is needed for correctness. And it shouldn't affect consumers of the feed at all, except for humans looking at the feed and the app at the same time. So I think I don't want to make it more specific.

Reply:
I guess I agree that it doesn't make sense for these two different values to line up all the time, but I'm just thinking that from an architectural and overall legibility perspective it would make sense to put some very high-level guardrails around how much this feed can differ from the Glides app view. I don't want us to end up in a situation where Glides is the only application doing light rail trip assignments but it's actually doing multiple radically different sets of trip assignments for different purposes.

Reply:
Another way that Glides might do different matching in-app vs this feed: We could consider showing inferred matches (where inspectors haven't entered data) in this feed, but consider those trains unmatched in-app, in order to prompt inspectors to manually fill out the trip and to avoid showing operator assignments we're less certain of.

(We aren't currently planning to do this, but it's something we've talked about and I wouldn't want the spec to prohibit it.)